lib/valueflow.cpp

```diff
@@ -865,7 +865,7 @@ static void setTokenValue(Token* tok, ValueFlow::Value value, const Settings* se
                     (value1.varId == value2.varId && value1.varvalue == value2.varvalue)) {
                     ValueFlow::Value result(0);
                     result.condition = value1.condition ? value1.condition : value2.condition;
-                    result.setInconclusive(value1.isInconclusive() | value2.isInconclusive());
+                    result.setInconclusive(value1.isInconclusive() || value2.isInconclusive());
```

Comment:
isInconclusive can be easily inlined here. I don't like the clang warning here.

Reply:
I think it has to do that the bitwise operator has precedence over the logical ones so if you would use `!` in there the result would not be expected. Still this just looks like a typo to me.

Reply:
That would have been unexpected to me.. but `!` still has higher precedence than both the logical and bitwise and.

It is not a typo in our code to use bitwise or/and. We do that intentionally sometimes. I am against this.

Reply:
I disabled it for now.